tests/Resources/ti.geolocation.test.js

```diff
@@ -303,53 +302,104 @@ describe.windowsBroken('Titanium.Geolocation', function () {
 		});
 	});
 
-	it('#forwardGeocoder()', function (finish) {
-		this.timeout(6e4); // 60 sec
+	describe('#forwardGeocoder()', () => {
+		it('is a Function', () => should(Ti.Geolocation.forwardGeocoder).be.a.Function());
 
-		should(Ti.Geolocation.forwardGeocoder).be.a.Function();
-		Ti.Geolocation.forwardGeocoder('440 N Bernardo Ave, Mountain View', function (data) {
-			try {
-				should(data).have.property('success').which.is.a.Boolean();
-				should(data.success).be.be.true();
-				should(data).have.property('code').which.is.a.Number();
-				should(data.code).be.eql(0);
-				should(data.latitude).be.approximately(37.387, 0.004); // iOS: 37.38605, Windows: 37.3883645, Android: 37.3909049
-				should(data.longitude).be.approximately(-122.065, 0.02); // Windows: -122.0512682, iOS: -122.08385, Android: -122.0472468
+		it('works via callback argument', function (finish) {
+			this.timeout(6e4); // 60 sec
+
+			Ti.Geolocation.forwardGeocoder('440 N Bernardo Ave, Mountain View', function (data) {
+				try {
+					should(data).have.property('success').which.is.a.Boolean();
+					should(data.success).be.be.true();
+					should(data).have.property('code').which.is.a.Number();
+					should(data.code).be.eql(0);
+					should(data.latitude).be.approximately(37.387, 0.004); // iOS: 37.38605, Windows: 37.3883645, Android: 37.3909049
+					should(data.longitude).be.approximately(-122.065, 0.02); // Windows: -122.0512682, iOS: -122.08385, Android: -122.0472468
+				} catch (err) {
+					return finish(err);
+				}
 				finish();
-			} catch (err) {
-				finish(err);
-			}
+			});
+		});
+
+		it('works via Promise return value', function (finish) {
+			this.timeout(6e4); // 60 sec
+
+			Ti.Geolocation.forwardGeocoder('440 N Bernardo Ave, Mountain View').then(data => {
+				try {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.geolocation.test.js;
  Line: 330;
-->

- :warning: tests/Resources/ti.geolocation.test.js line 330 – Each then() should return a value or throw (promise/always-return)